src/app/_contexts/Popup.context.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+import React, { createContext, useContext, useState } from "react";
+import {
+  ConcedePopup,
+  DefaultPopup,
+  MulliganPopup,
+} from "../_components/_sharedcomponents/Popup/Popup.types";
+
+// TODO: SelectCardsPopup | LeaderAbilityPopup | DrawPopup | SelectCardsWithAspectPopup
+export type PopupData = DefaultPopup | MulliganPopup | ConcedePopup;
+
+export type PopupType = PopupData["type"];
+
+export type PopupDataMap = {
+  default: Omit<DefaultPopup, "type">;
+  mulligan: Omit<MulliganPopup, "type">;
+  concede: Omit<ConcedePopup, "type">;
+};
+
+interface PopupContextProps {
+  type?: PopupType;
+  data?: PopupData;
+  openPopup: <T extends PopupType>(type: T, data: PopupDataMap[T]) => void;
+  closePopup: () => void;
+}
+
+const PopupContext = createContext<PopupContextProps | undefined>(undefined);
+
+export const PopupProvider: React.FC<{ children: React.ReactNode }> = ({
```

Comment:
Do we have support for multiple popups here or is that still WIP?

Reply:
Not yet. But refactoring the code to handle multiple popups will not be that struggling